totp/totp_test.go

```diff
@@ -0,0 +1,87 @@
+package totp
+
+import (
+	"testing"
+	"time"
+
+	"github.com/libsv/go-bk/bip32"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTotpService(t *testing.T) {
+	const a_xprivStr = "xprv9s21ZrQH143K4BNj7w6PX1kTzUDMrKGc9VUEHhguqdbdyPVdP6t6NJNNMxuksetwDCFiauEvwNtNyt5xkXPS6eDtvf7e1GAcDERdoeApfGc"
+	const b_xprivStr = "xprv9s21ZrQH143K2yLE69dKjrRmHE5atymAuVzqWj7gZcBDdhSZQcGZiFmMBwpXDC36bvMV398HXwVK4UUCwB5oddU5dU93QKD7JFyyizYwLDp"
+
+	t.Run("Passcode has the exact number of digits", func(t *testing.T) {
```

Comment:
Suggestion:
It will be great if we will have all paths covered - I mean that we can reach all the errors from GenerateTotp function

Reply:
Errors that can we returned from `GenerateTotp()` are from external packages like `github.com/libsv/go-bk/bip32` or `github.com/pquerna/otp`.

Do you suggest testing specific cases for these packages? IMO it's not a path we should go.

Reply:
Ok, so let's stay with all paths that we control directly, like missing xpriv etc.

Reply:
I believe you talk about [these tests](https://github.com/bitcoin-sv/spv-wallet-go-client/pull/212/files#diff-c0e64ef1406ebe9951fd601684c57b0c0e7a2ae81be1d2e9b1d9581727a88def)

NOTE: I added them after your comment